pinot-core/src/main/java/org/apache/pinot/core/plan/CombinePlanNode.java

```diff
@@ -175,35 +179,54 @@ public List<Operator> callJob() {
       }
     }
 
-    if (_streamObserver != null && QueryContextUtils.isSelectionOnlyQuery(_queryContext)) {
-      // Streaming query (only support selection only)
-      return new StreamingSelectionOnlyCombineOperator(operators, _queryContext, _executorService, _streamObserver);
-    }
-    if (QueryContextUtils.isAggregationQuery(_queryContext)) {
-      if (_queryContext.getGroupByExpressions() == null) {
-        // Aggregation only
-        return new AggregationCombineOperator(operators, _queryContext, _executorService);
+    if (_streamObserver != null) {
```

Comment:
This is not related to / introduced by this PR but if possible please consider splitting this into  1-2 functions. The long if-else chain looks unreadable.

Reply:
yes i think we can follow up with separate PR, in fact my next PR will include changes to this file so will look into it next